java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowStreamWriter.java

```diff
@@ -68,18 +69,18 @@ public ArrowStreamWriter(VectorSchemaRoot root, DictionaryProvider provider, Wri
    * Write an EOS identifier to the WriteChannel.
    *
    * @param out Open WriteChannel with an active Arrow stream.
+   * @param option IPC write option
    * @throws IOException on error
    */
-  public void writeEndOfStream(WriteChannel out) throws IOException {
-    if (option.write_legacy_ipc_format) {
-      out.writeIntLittleEndian(0);
-    } else {
-      out.writeLongLittleEndian(0);
+  public static void writeEndOfStream(WriteChannel out, IpcOption option) throws IOException {
+    if (!option.write_legacy_ipc_format) {
+      out.writeIntLittleEndian(MessageSerializer.IPC_CONTINUATION_TOKEN);
```

Comment:
Also apply this change to ArrowFileWriter#endInternal and probably remove MessageSerializer#writeLongLitterEndian since it’s no use anymore?

Reply:
good catch, I forgot that also writes EOS